### PR TITLE
Encryption 

### PR DESCRIPTION
**Pull request by skwarq**

### CODE REVIEW

**Comment by ziobron** [comment]
@skwarq, we do not have your data in the Coders School database.
1. Please fill in this form https://forms.gle/rkzLsMUSBdnGQgdM9
2. Edit this PR to run this automation once again (you can just add a space in the description or title).
3. Wait one minute. If any label was added to this PR (like Standard, Premium, Podstawy C++, STL) then everything is fine :)
4. If your solution passed, but nothing happens, modify your code (eg. add an empty line) to run the workflow again.

If you don't know what to do - ask on Discord and provide a link to this PR.

File: encrypt/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build/
```
Comment (emttiew) [inline encrypt/.gitignore:1]: We already have .gitignore file with excluded build folder.
https://github.com/coders-school/password-manager/blob/main/.gitignore

File: encrypt/uTests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+set(TEST_NAME ${PROJECT_NAME}-UT)
+
+FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY "https://github.com/google/googletest.git"
+    GIT_TAG release-1.11.0
+)
+
+FetchContent_MakeAvailable(googletest)
```
Comment (emttiew) [inline encrypt/uTests/CMakeLists.txt:11]: GTest is already configured here: https://github.com/coders-school/password-manager/blob/main/source/test/CMakeLists.txt

File: encrypt/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        add_dependencies(OpenSSL::Crypto OpenSSL)
+    endif()
+endif()
+
+add_library(encrpyt ${project_src})
+target_link_libraries(encrpyt PUBLIC OpenSSL::SSL OpenSSL::Crypto)
+target_include_directories(encrpyt PUBLIC "../")
+target_compile_options(encrpyt PRIVATE -Wall -Wextra -Wpedantic -Werror)
+
+# if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR MODERN_CMAKE_BUILD_TESTING) AND(BUILD_TESTING))
```
Comment (emttiew) [inline encrypt/CMakeLists.txt]: We shouldn't push commented code.

File: encrypt/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    enable_testing()
```
Comment (emttiew) [inline encrypt/CMakeLists.txt:60]: Already defined here: https://github.com/coders-school/password-manager/blob/main/CMakeLists.txt#L11

File: encrypt/CMakeLists.txt (hunk 3 above)
Comment (skwarq) [reply encrypt/CMakeLists.txt]: Yes, my fault, it was just for testing. Uncommented

File: encrypt/uTests/CMakeLists.txt (hunk 2 above)
Comment (skwarq) [reply encrypt/uTests/CMakeLists.txt:11]: Encrypt was prepared as a standalone lib. You can simply use in another project as a submodule. I can bind this to the project, but I don't think it's a better option.

File: encrypt/uTests/symCrypt/AES256CBCTests.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  std::vector<unsigned char> aesKey, iv;
+  generateAESKeyAndIV(aesKey, iv);
+  AES256CBC aes({iv.data(), iv.size()});
+  std::string plainText{"Hello word!"};
+
+  std::vector<unsigned char> cipherText(16);
+  std::span cipherBuff = {cipherText.data(), cipherText.size()};
+  aes.encrypt(
+      {aesKey.begin(), aesKey.end()},
+      {reinterpret_cast<unsigned char *>(plainText.data()), plainText.size()},
```
Comment (emttiew) [inline encrypt/uTests/symCrypt/AES256CBCTests.cpp:31]: Is it necessary to use `reinterpret_cast`? `plainText.data()` is not enough?

File: encrypt/uTests/symCrypt/AES256CBCTests.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    throw std::runtime_error("Error generating AES key");
+  }
+
+  iv.resize(16);  // IV size for AES CBC mode
+  if (RAND_bytes(iv.data(), iv.size()) != 1) {
+    throw std::runtime_error("Error generating IV");
+  }
+}
+
+TEST(AES256CBCTests, simpleEncryptDecrypt) {
```
Comment (emttiew) [inline encrypt/uTests/symCrypt/AES256CBCTests.cpp:21]: This UT name is ok, though we could introduce a more standardized naming convention. Consider adopting a structure like "should" for expected behavior or "when...then" for specific conditions and outcomes.

**Review by emttiew — APPROVED** [review]

File: encrypt/uTests/symCrypt/AES256CBCTests.cpp (hunk 5 above)
Comment (skwarq) [reply encrypt/uTests/symCrypt/AES256CBCTests.cpp:31]: Yes is necessary. plainText.data() returns char*.

**Pull request merged**